parquet-benchmarks/src/main/java/org/apache/parquet/benchmarks/ReadBenchmarks.java

```diff
@@ -103,4 +111,61 @@ public void read1MRowsDefaultBlockAndPageSizeGZIP(Blackhole blackhole)
   {
     read(file_1M_GZIP, ONE_MILLION, blackhole);
   }
+
+  private void concurrentRead(Path f, int numThreads, Blackhole blackhole) throws IOException, InterruptedException
+  {
+    List<Thread> threads = new ArrayList<>();
+    // Read the file metadata for block assignments afterwards.
+    ParquetMetadata footer = ParquetFileReader.readFooter(configuration, f, NO_FILTER);
+    List<BlockMetaData> blocks = footer.getBlocks();
+    int totalRowBlocks = blocks.size();
+    if (totalRowBlocks < numThreads) {
+      numThreads = totalRowBlocks;
+    }
+
+    // Assign blocks to each thread
+    int index = 0;
+    int slop = totalRowBlocks % numThreads;
+    for (int i = 0; i < numThreads; i++) {
+      int numRows = (totalRowBlocks - slop) / numThreads;
+      // Try load balancing
+      if (i < slop) {
+        ++numRows;
+      }
+      List<BlockMetaData> subBlocks = blocks.subList(index, index + numRows);
+      FileMetaData meta = footer.getFileMetaData();
+      Thread t = new ParquetScanner(f, meta, subBlocks, meta.getSchema().getColumns(), blackhole);
+      t.start();
+      threads.add(t);
+      index += numRows;
+    }
+
+    for (Thread t: threads) {
+      t.join();
+    }
+  }
+
+  @Benchmark
+  @Threads(1)
+  public void readUncompressed16Thread(Blackhole blackhole)
```

Comment:
Could you remove the specific thread count (16) from the name of these methods? You could use Multi instead (e.g., readUncompressedMultiThread).